### PR TITLE
fix for mmap file creation

### PR DESCRIPTION
**Pull request by ewoolsey**

### CODE REVIEW

File: src/generic_storage/mmap_vec.rs [hunk 1]
```diff
@@ -63,7 +62,12 @@ impl<T: Pod> MmapVec<T> {
     /// # Safety
     /// Same requirements as `restore`
     pub unsafe fn restore_from_path(file_path: impl AsRef<Path>) -> color_eyre::Result<Self> {
-        let file = OpenOptions::new().read(true).write(true).open(file_path)?;
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .truncate(false)
```
Comment (Dzejkop) [inline src/generic_storage/mmap_vec.rs:69]: Won't this overwrite file contents?

Comment (ewoolsey) [reply src/generic_storage/mmap_vec.rs:69]: `.truncate(false)` prevents that.

File: src/generic_storage/mmap_vec.rs [hunk 2]
```diff
@@ -245,6 +249,202 @@ mod tests {
...
+            size_of::<u32>() * 8 + META_SIZE
+        );
+
+        let storage: MmapVec<u32> = unsafe { MmapVec::restore(f.reopen().unwrap()).unwrap() };
+        assert_eq!(storage.capacity, 8);
+        assert_eq!(
+            std::fs::metadata(&file_path).unwrap().len() as usize,
+            size_of::<u32>() * 8 + META_SIZE
+        );
+    }
```
Comment (ewoolsey) [inline src/generic_storage/mmap_vec.rs:418]: @Dzejkop This test should demonstrate that it works correctly

File: src/generic_storage/mmap_vec.rs (hunk 1 above)
Comment (Dzejkop) [reply src/generic_storage/mmap_vec.rs:69]: oh, I'm blind I thought this was `true`

File: src/generic_storage/mmap_vec.rs (hunk 2 above)
Comment (Dzejkop) [reply src/generic_storage/mmap_vec.rs:418]: yep, looks good

**Review by Cemg10 — COMMENTED** [review]
Muy bajos

**Pull request merged**